documentation/tutorials/get-started-with-ash-and-phoenix.md

```diff
@@ -0,0 +1,505 @@
+
+## Goals
+
+In this guide we will:
+
+1. Create a new Phoenix project
+2. Setup Ash, AshPhoenix and AshPostgres as dependencies
+3. Create a basic `Blog.Post` resource
+4. Create and migrate the database
+5. Learn how to interact with your resource
+6. Integrate a minimal Phoenix LiveView with Ash
+
+## Things You May Want to Read First
+
+- [Install Elixir](https://elixir-lang.org/install.html)
+- [Phoenix - Up and Running Guide](https://hexdocs.pm/phoenix/up_and_running.html)
+- [Philosophy Guide](/documentation/tutorials/philosophy.md)
+
+## Requirements
+
+If you want to follow along yourself, you will need the following things:
+
+1. Elixir (1.12 or later) and Erlang (22 or later) installed
+2. PostgreSQL installed
+3. A text editor
+4. A terminal to run the examples
+
+## Setup
+
+### Create a New Phoenix Project
+
+_This section is based on the [Phoenix installation docs](https://hexdocs.pm/phoenix/installation.html). For more details go there._
+
+First we need to install the Phoenix project generator, then we'll run the generator to create our new project.
+
+**NOTE: DO NOT run `mix ecto.create`, (as it asks you to) we will do this the Ash way later.**
+
+```bash
+# install Phoenix project generator
+$ mix archive.install hex phx_new
+
+# generate Phoenix project
+$ mix phx.new my_ash_phoenix_app
+
+# cd into project
+$ cd my_ash_phoenix_app
+```
+
+### Add Dependencies
+
+We now need to add Ash, AshPhoenix and AshPostgres to our Phoenix project. We need to add the dependencies to the `deps` function in our `mix.exs`. We'll also need to add dependencies to our `.formatter.exs` to ensure consistent formatting when using `mix format`.
+
+```elixir
+# mix.exs
+
+  def deps do
+    [
+      {:phoenix, "~> 1.6.15"},
+      # ...
+      {:ash, "~> 2.6.2"},
+      {:ash_postgres, "~> 1.3.10"},
+      {:ash_phoenix, "~> 1.2.5"}
+    ]
+  end
+```
+
+```elixir
+# .formatter.exs
+[
+  import_deps: [:ecto, :phoenix, :ash, :ash_phoenix, :ash_postgres],
+  inputs: ["*.{ex,exs}", "priv/*/seeds.exs", "{config,lib,test}/**/*.{ex,exs}"],
+  subdirectories: ["priv/*/migrations"]
+]
+```
+
+Now in the terminal install these new dependencies.
+
+```bash
+$ mix deps.get
+```
+
+### Use `AshPostgres.Repo` and Create the Database
+
+We need to swap `Ecto.Repo` for `AshPostgres.Repo`. `AshPostgres.Repo` enriches your repo with additional AshPostgres specific behaviour, but is essentially a thin wrapper around `Ecto.Repo`.
+To use `AshPostgres.Repo` change your repo module to look like this:
+
+```elixir
+# lib/my_ash_phoenix_app/repo.ex
+
+defmodule MyAshPhoenixApp.Repo do
+  use AshPostgres.Repo, otp_app: :my_ash_phoenix_app
+
+  # Installs Postgres extensions that ash commonly uses
+  def installed_extensions do
+    ["uuid-ossp", "citext"]
+  end
+end
+```
+
+After swapping `AshPostgres.Repo` in, you can now create the database using:
+
+```
+$ mix ash_postgres.create
```

Comment:
I got the error about needing to set the config for backwards compatibility at this step - it happens when compiling Ash.

```
==> ash
Compiling 333 files (.ex)
warning: * IMPORTANT *

The configuration `use_all_identities_in_manage_relationship` was not set.
It is defaulting to `true` for backwards compatibility.

This configuration must now be manually set, and it should be set to `false`.
If you have just started a new project, or haven't used `manage_relationship` yet, just set the following config:

config :ash, :use_all_identities_in_manage_relationship?, false

If you are currently using `manage_relationship`, please read https://github.com/ash-project/ash/issues/469 before proceeding

  lib/ash/actions/managed_relationships.ex:15: (module)
```

Maybe the Edit Config section needs to be moved up a bit.

Reply:
We don't really need to call `mix ash_postgres.create` until we migrate the database. So I'll move it down to that section.

Reply:
[8a2b35f](https://github.com/ash-project/ash_phoenix/pull/70/commits/8a2b35f9bac7edef885ce5a6f3337fa2d68dfa38)